tests/basic_correctness/test_basic_correctness.py

```diff
@@ -118,11 +120,11 @@ def test_models_distributed(
 
     if model == "meta-llama/Llama-3.2-1B-Instruct" and distributed_executor_backend == "ray" and attention_backend == "" and test_suite == "L4":  # noqa
         # test Ray Compiled Graph
-        os.environ['VLLM_USE_RAY_SPMD_WORKER'] = "1"
-        os.environ['VLLM_USE_RAY_COMPILED_DAG'] = "1"
+        monkeypatch.setenv("VLLM_USE_RAY_SPMD_WORKER", "1")
```

Comment:
actually, one final thing, can you move this within `monkeypatch.context` instead of just setenv here?

https://docs.pytest.org/en/stable/reference/reference.html#pytest.MonkeyPatch.context

We probably don't want to have any side-effects here.